### ISSUE TITLE
Suppress US in office address fields

### ISSUE THREAD

**Issue by GaryRidgway** [uiowa/uiowa#4569]
Users have reported that when they create an address for an office that they would like to remove the United states from it. The current workaround is that they are using the phone number field to list their office

example here:
https://org-iowajpec.prod.drupal.uiowa.edu/node/21/edit

## Proposed solution

Enable the [address display module](https://www.drupal.org/project/address_display) and use it to allow users to check on or off the display of the country.

**Comment by briand44** [comment]
possible solutions in no order:

1. Don't show country if country is US.
2. Add checkbox so users can choose wether or not to show country on display.
3. Write code to evaluate values entered and if not all fields have data don't output country (e.g only adding building name not city, state, zip).

**Comment by joewhitsitt** [comment]
Something to keep in mind, which is probably part of the reason Sean and I are wary of adding more complexity to this functionality, is that we are already doing some crazy stuff to person contact information already in `_sitenow_people_person_form_submit`

**Comment by joewhitsitt** [comment]
Also, FWIW there is an issue upstream being worked on https://www.drupal.org/project/address/issues/2753899 that addresses [potential solution #1](https://github.com/uiowa/uiowa/issues/4569#issuecomment-1027221303) for a per field instance basis.

**Comment by dnkemper** [comment]
> https://org-iowajpec.prod.drupal.uiowa.edu/node/21/edit

Yeah.. that was why I suggested the boolean field. A simple way to show/hide. Also, https://www.drupal.org/project/address/issues/2753899  looks abandoned... uses php 5.5 and D8 too.. bummer

**Comment by dnkemper** [comment]
Disregard the steps. This is blocked.

**Comment by joewhitsitt** [comment]
~~I don't think https://github.com/uiowa/uiowa/issues/4569#issuecomment-1027265827 is working anymore~~ it is working. i had forgotten what it was doing...

**Comment by joewhitsitt** [comment]
The address field template is expecting a value and doesn't have any kind of if check around it which causes issues if there is a preprocess unset on it.

docroot/modules/contrib/address/templates/address-plain.html.twig

**Comment by joewhitsitt** [comment]
> Actually, the patch from https://www.drupal.org/project/address/issues/2753899#comment-12982080 works pretty well. I didn't encounter the issues the following comments reported. If we went with it, we should probably re-save all of our address fields (across splits) to make sure the functionality is registered correctly whether or not the option is used.

https://github.com/uiowa/uiowa/pull/5934#issuecomment-1332739983

**Comment by briand44** [comment]
Nursing asked about this as well.

**Comment by joewhitsitt** [comment]
This is coming up now for the contact block where minimal info has been entered for the address but otherwise the need to display the United States with it is not desired.

From https://github.com/uiowa/uiowa/pull/8115

<img width="356" height="294" alt="Image" src="https://github.com/user-attachments/assets/d0184fd7-de5a-4d9d-b95d-36dc84e362fa" />

**Comment by joewhitsitt** [comment]
# Grooming Conversation

**Quadrant:** 

Complicated

**Discussion:**

- Past attempt to hide via class/css if country code is US. https://github.com/uiowa/uiowa/pull/4752
- Upstream issue has a patch but has failed to get traction and commitment from maintainers - https://www.drupal.org/project/address/issues/2753899
- Might be some preprocessing available to unset or hide right before render
- Need to handle beyond the person content type now that the addressfield is being used elsewhere
- Consider the editorial style guide to determine to what granularity this should be controlled (platform wide, content wide, site wide, instance wide). Easiest would be platform wide.
- Spot check styles to make sure it looks good

**Proposed solution or definition of done:**

- Implement a solution the team is happy with that remains compatible with potential future updates from the module.
